File: eureka/android/library/src/main/java/com/lifeplus/Util/Global.java
package com.lifeplus.Util;

public class Global {
    public static final boolean _onSimulator = false;
    private static boolean _debugMode = true;
    private static String _appVersion = "1.0.0";         // TODO: read from manifest

    private static int _userId = 0;
    private static String _watchMSN = "";

    private static int _userIdForScan = 0;
    private static String _watchMsnForScan;
    private static String _currentStatus = null;

    public static void setUserIdForScan(int pUserIdForScan) {
        _userIdForScan = pUserIdForScan;
    }

    public static int getUserIdForScan() {
        return _userIdForScan;
    }

    public static void setWatchMSNForScan(String pWatchMsnForScan) {
        _watchMsnForScan = pWatchMsnForScan;
    }

    public static String getWatchMSNForScan() {
        return _watchMsnForScan;
    }

    public static void setUserId(int pUserId) {
        _userId = pUserId;
    }

    public static int getUserId() {
        return _userId;
    }

    public static String getWatchMSN() {
        return _watchMSN;
    }

    public static void setWatchMSN(String pWatchMSN) {
        _watchMSN = pWatchMSN;
    }

    public static boolean isDebugMode() {
        return _debugMode;
    }

    public static void setDebugMode(boolean debugMode) {
        _debugMode = debugMode;
    }

    public static String getAppVersion() {
        return _appVersion;
    }

    public static void setAppVersion(String appVersion) {
        _appVersion = appVersion;
    }

    public static String getCurrentStatus() {
        return _currentStatus;
    }

    public static void setCurrentStatus(String pCurrentStatus) {
        Global._currentStatus = pCurrentStatus;
    }

    public static void moveConfirmScanUser() {
        _userId = _userIdForScan;
        _watchMSN = _watchMsnForScan;
        _userIdForScan = 0;
        _watchMsnForScan = "";
    }
}

File: eureka/ios/nativeLib/nativeLib/Utils/Global.swift
//
//  Global.swift
//  eureka
//
//  Created by work on 04/02/21.
//

import Foundation

public class Global {
  public static let autoCalibrationOn = false
  public static let onSimulator:Bool = false
  private static var _debugMode:Bool = true
  private static var _autoMeasure:Bool = false

  private static var _userId:Int = 0
  private static var _watchMSN:String = ""

  private static var _userIdForScan:Int = 0
  private static var _watchMsnForScan:String = ""

  public static func setUserIdForScan(pUserIdForScan:Int) {
      _userIdForScan = pUserIdForScan
  }

  public static func getUserIdForScan() -> Int {
      return _userIdForScan
  }

  public static func setWatchMSNForScan(pWatchMsnForScan: String) {
      Global._watchMsnForScan = pWatchMsnForScan
  }

  public static func getWatchMSNForScan() -> String {
    return _watchMsnForScan
  }

  public static func setUserId(pUserId: Int) {
      _userId = pUserId
  }

  public static func getUserIdStr() -> String {
      return "\(_userId)"
  }

  public static func getUserId() -> Int {
      return _userId
  }

  public static func getWatchMSN() -> String {
      return _watchMSN
  }

  public static func isDebugMode() -> Bool {
      return _debugMode
  }

  public static func setDebugMode(debugMode: Bool) {
      Global._debugMode = debugMode
  }

  public static func isAutoMeasure() -> Bool {
        return _autoMeasure
  }

  public static func setAutoMeasure(_autoMeasure: Bool) {
      Global._autoMeasure = _autoMeasure
  }

  public static func moveConfirmScanUser() {
    _userId = (_userIdForScan == 0) ? _userId : _userIdForScan
    _watchMSN = (_watchMsnForScan == "") ? _watchMSN : _watchMsnForScan

    _userIdForScan = 0
    _watchMsnForScan = ""
  }
    
    public static func getWatchMSNPure() -> String {
        let value = !getWatchMSN().isEmpty ? getWatchMSN() : getWatchMSNForScan()
        if let splitIndex = value.lastIndex(of: "-") {
            return String(value[value.index(after: splitIndex)...])
        } else {
            return value
        }
    }
}
